### ISSUE TITLE
error while installing Exchange

### ISSUE THREAD

**Issue by JestyDS** [MicrosoftDocs/OfficeDocs-Exchange#1882]
[Enter feedback here]
Performing Microsoft Exchange Server Prerequisite Check

.\setup.exe : The following error was generated when "$error.Clear(); 
At line:2 char:1
+ .\setup.exe /mode:Install /role:Mailbox /OrganizationName:Contoso /Ia ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (The following e...error.Clear(); :String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
          if (Get-Service MpsSvc* | ?{$_.Name -eq 'MpsSvc'})
          {
            Set-Service MpsSvc -StartupType Automatic
            Start-SetupService -ServiceName MpsSvc
          }
        "
 was run: "Microsoft.PowerShell.Commands.ServiceCommandException: Service 'Windows Defender Firewall (MpsSvc)' 
cannot be configured due to the following error: Access is denied ---> System.ComponentModel.Win32Exception: Access 
is denied
   --- End of inner exception stack trace ---".
 Configuring Prerequisites
The Net.Tcp Port Sharing service must be set to Automatic startup and started before Setup can continue.
This computer isn't part of an Active Directory domain.
The user isn't logged on to an Active Directory domain.
This computer doesn't belong to a valid Active Directory site. Check the site and subnet definitions.
This computer requires .NET Framework 4.8 (https://support.microsoft.com/kb/4503548).
You must be a member of the 'Organization Management' role group or a member of the 'Enterprise Admins' group to 
continue.
You must use an account that's a member of the Organization Management role group to install or upgrade the first 
Mailbox server role in the topology.
You must use an account that's a member of the Organization Management role group to install the first Client Access 
server role in the topology.
You must use an account that's a member of the Organization Management role group to install the first Client Access 
server role in the topology.
You must use an account that's a member of the Organization Management role group to install or upgrade the first 
Mailbox server role in the topology.
You must use an account that's a member of the Organization Management role group to install or upgrade the first 
Client Access server role in the topology.
You must use an account that's a member of the Organization Management role group to install the first Mailbox 
server role in the topology.
Setup encountered a problem while validating the state of Active Directory: Active Directory server  is not 
available. Error message: Active directory response: The LDAP server is unavailable.  See the Exchange setup log for 
more information on this error.
Either Active Directory doesn't exist, or it can't be contacted.
The Windows component Server-Gui-Mgmt-Infra isn't installed on this computer and needs to be installed before 
Exchange Setup can begin.
An unsupported operating system was detected. Exchange Server 2016 Mailbox and Edge Transport server roles support 
Windows Server 2012 or later.
Visual C++ 2013 Redistributable Package is a required component. Please install the required binaries and re-run the 
setup. Use URI https://www.microsoft.com/download/details.aspx?id=40784 to download the binaries.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e45c04a0-138f-8241-0726-01fb09c3dfeb
* Version Independent ID: 730303a8-9c6f-472c-79f6-96453dea3013
* Content: [Exchange dev/test environment in Azure](https://docs.microsoft.com/en-us/exchange/plan-and-deploy/deploy-new-installations/create-azure-test-environments?view=exchserver-2016#feedback)
* Content Source: [Exchange/ExchangeServer/plan-and-deploy/deploy-new-installations/create-azure-test-environments.md](https://github.com/MicrosoftDocs/OfficeDocs-Exchange/blob/public/Exchange/ExchangeServer/plan-and-deploy/deploy-new-installations/create-azure-test-environments.md)
* Product: **exchange-server-it-pro**
* Technology: ****
* GitHub Login: @mattpennathe3rd
* Microsoft Alias: **v-mapenn**

**Comment by haflidif** [comment]
Hey JestyDS, are you running the PowerShell window with elevated permission (run as administrator), just checking the basics :-)

**Comment by haflidif** [comment]
Hey JestyDS, Is your exVM joined to the domain of adVM and are you able to communicate between those machines ? 
As the Error log states as well 
This computer isn't part of an Active Directory domain.
The user isn't logged on to an Active Directory domain.
This computer doesn't belong to a valid Active Directory site. Check the site and subnet definitions.

**Comment by haflidif** [comment]
As well as I see you are missing some prereqs like .Net 4.8 and Visual C++ 2013  Redistrubatble Package, this needs to be installed on the server before installing Exchange 2016.

**Comment by JestyDS** [comment]
Sure. I saw them as per the doc it says to add the DNS name on the AD server. When I connected the exchange server to the domain I am no longer able to login the server. I am using adminstrator shell on the server

**Comment by JestyDS** [comment]
I saw the exchange server was a workgroup server hence added it to the domain . Now I'm unable to login to the server after server restart.

**Comment by haflidif** [comment]
Do you see the Exchange server in Active Directory under Computers ?, just to check if you've successfully joined the machine to the Domain, as well you now need to login to the server using domain credentials, and to manage the server using Powershell is maybe blocked by the  Domain firewall profile

You need to open exVNET and set the DNS there to the private Ip Address on the domain controller if you have them in separate vNets.

I created this using the same vNet but separate subnets :) one subnet for the AD and one for the Exchange server.

**Comment by JestyDS** [comment]
I am able to see the server under computers in ad. As you had mentioned this could be a ip config issue. I will try creating a different subnet and see if it works. Thank you

**Comment by haflidif** [comment]
okay, so the server is at least joined to the domain, are you connecting to the exVM using the domain credentials, not the local admin credentials?

**Comment by haflidif** [comment]
basically user@contoso.corp or something instead of the exVM local Admin.
This Credentials -> $cred=Get-Credential -Message "Type the name and password of the local administrator account for adVM."

Not this credentials: $cred=Get-Credential -Message "Type the name and password of the local administrator account for exVM."

**Comment by JestyDS** [comment]
Oh my bad. I was using the local admin account (exvm) for logging onto the server.

**Comment by haflidif** [comment]
No problem, hope this will solve your problem, remember to install .net 4.8 and  Visual C++ 2013 Redistrubatble Package before running the Exchange Setup :), feel free to vote for my answers if they helped you.
Good luck going forward.

**Comment by JestyDS** [comment]
I had logged onto the Exchange server using the Domain account and had installed the prerequisites but i still get an error while running the Exchange setup
 Configuring Prerequisites
.\setup.exe : The following error was generated when "$error.Clear(); 
At line:2 char:1
+ .\setup.exe /mode:Install /role:Mailbox /OrganizationName:Contoso /Ia ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (The following e...error.Clear(); :String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
          if (Get-Service MpsSvc* | ?{$_.Name -eq 'MpsSvc'})
          {
            Set-Service MpsSvc -StartupType Automatic
            Start-SetupService -ServiceName MpsSvc
          }
        "
 was run: "Microsoft.PowerShell.Commands.ServiceCommandException: Service 'Windows Defender Firewall (MpsSvc)' 
cannot be configured due to the following error: Access is denied ---> System.ComponentModel.Win32Exception: Access 
is denied
   --- End of inner exception stack trace ---".
 Prerequisite AnalysisSetup will prepare the organization for Exchange Server 2016 by using 'Setup /PrepareAD'. No Exchange Server 201
3 roles have been detected in this topology. After this operation, you will not be able to install any Exchange Server 2013 roles.
Setup will prepare the organization for Exchange Server 2016 by using 'Setup /PrepareAD'. No Exchange Server 2010 roles have been dete
cted in this topology. After this operation, you will not be able to install any Exchange Server 2010 roles.
The Net.Tcp Port Sharing service must be set to Automatic startup and started before Setup can continue.
The Windows component Server-Gui-Mgmt-Infra isn't installed on this computer and needs to be installed before Exchange Setup can 
begin.
An unsupported operating system was detected. Exchange Server 2016 Mailbox and Edge Transport server roles support Windows Server 
2012 or later.


The Exchange Server setup operation didn't complete.  More details can be found in ExchangeSetup.log located in the <SystemDrive>:\Exc
hangeSetupLogs folder.

**Comment by haflidif** [comment]
Are you installing the Exchange Server on Server Core ? hence the Server-Gui-Mgmt-Infra windows feature error and if that's correct the Server OS you are using for your exVM is not supported, it needs full version of Windows Server 2012, 2012r2, 2016 - Standard or Datacenter

**Comment by JestyDS** [comment]
Hi haflidif

I am using a Windows server 2019 datacenter edition. Does it not support?

**Comment by JestyDS** [comment]
My bad Exchange 2016 does not support Windows server 2019 ..

**Comment by haflidif** [comment]
No that's correct Exchange 2016 doesn't support Server 2019, try at least windows server 2016 :) 
Good luck going forward.

**Comment by mattpennathe3rd** [comment]
@JestyDS @haflidif I'm going to assume this one is not a documentation problem and will close this issue now, but please feel free to open a new issue if you find there's any error or omission in the documentation. Just so you know, this repo is not checked by Microsoft support staff and I'm only equipped to help with documentation problems here. (I'm only a Technical Writer. 😄 )

**Comment by eschrader** [comment]
If you are using Windows Server 2019, just download Exchange 2019 from my.visualstudio.com w/ CU5 (contains entire installer). Install pre-reqs and there shouldnt be any issues.
